File: frontend/src/types/styleRecommendation.ts
export interface StyleSuggestion {
  clothing_id: string
  name: string
  category: string
  primary_color: string
  price: number
  currency: string
  slot: string
  reason: string
}

export interface StyleRecommendation {
  anchor_clothing_id: string
  anchor_name: string
  anchor_color: string
  occasion: string | null
  suggestions: StyleSuggestion[]
  note: string
}

export interface StyleRecommendationRequest {
  clothing_id: string
  occasion?: string
}

File: backend/app/schemas/style_recommendation.py
import uuid

from pydantic import BaseModel


class StyleRecommendationRequest(BaseModel):
    clothing_id: uuid.UUID
    occasion: str | None = None


class StyleSuggestion(BaseModel):
    clothing_id: uuid.UUID
    name: str
    category: str
    primary_color: str
    price: float
    currency: str
    slot: str
    reason: str


class StyleRecommendationResponse(BaseModel):
    anchor_clothing_id: uuid.UUID
    anchor_name: str
    anchor_color: str
    occasion: str | None
    suggestions: list[StyleSuggestion]
    note: str
